### PR TITLE
Issue #13999: Killed mutation for ScopeUtil

### PR DESCRIPTION
**Pull request by Rohanraj123**
Related #13999 

Mutation : 

https://github.com/checkstyle/checkstyle/blob/60267bb9848194182de1f6a0980999a5f5bcbe28/config/pitest-suppressions/pitest-utils-suppressions.xml#L12-L19

Diff Regression Config : https://raw.githubusercontent.com/Rohanraj123/Rohanraj123.github.io/main/my_check.html

Diff Regression Projects : https://raw.githubusercontent.com/Rohanraj123/Rohanraj123.github.io/main/ProjectsConfig

### CODE REVIEW

**Comment by Rohanraj123** [comment]
Github, generate report

**Comment by nrmancuso** [comment]
@Rohanraj123 few things:

- How did you build this dependency tree? What does it mean?
- How did you determine that we needed to generate a check regression report for NestedSwitchCheck?
- Why didn't we simply hardcode the surviving mutation that pitest found?

Also, we will need to run check regression testing on ALL checks that use this method (with a few different configurations):
```
➜  checkstyle ag -lQ "ScopeUtil.getSurroundingScope(" --ignore "*Test.java"
src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocVariableCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocMethodCheck.java
src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java

```

I suggest that you put this PR into draft mode while you work on these tasks.

**Comment by Rohanraj123** [comment]
@nrmancuso I am working on asked questions but you can please first confirm me that changes made was okay or not.

**Comment by nrmancuso** [comment]
> can you please first confirm me that changes made was okay or not

No one can answer this question until you have answered my questions above, and generated a report :) We have no way to know how your changes have affected all of the checks that use this method. I suggest you strictly follow steps at https://github.com/checkstyle/checkstyle/issues/12341.

**Comment by Rohanraj123** [comment]
> * How did you build this dependency tree? What does it mean?

I have updated dependency tree. Letme first explain what my understanding of dependency tree is : Which are other checks methods which depends on the specific method. If yes then I think now dependency tree must be understandable. And also Updated the checks needed to generate regression report over. 

> * How did you determine that we needed to generate a check regression report for NestedSwitchCheck?

Yeah I was wrong here I have updated checks

> * Why didn't we simply hardcode the surviving mutation that pitest found?

If you mean by adding suppressions then this is something which is not allowed.

**Comment by nrmancuso** [comment]
> I have updated dependency tree. Letme first explain what my understanding of dependency tree is …

Let me try to be more clear: why do we need this “dependency tree”? Where in https://github.com/checkstyle/checkstyle/issues/12341 is it advised to do this? 

> If you mean by adding suppressions then this is something which is not allowed.

Please carefully read though https://github.com/checkstyle/checkstyle/issues/12341. You need to first change the code to exactly what pitest is mutating to.

**Comment by Rohanraj123** [comment]
> Please carefully read though #12341. You need to first change the code to exactly what pitest is mutating to.

I have read #12341 . I have removed the unwanted material.

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java [hunk 1]
```diff
@@ -131,23 +131,21 @@ else if (isInInterfaceOrAnnotationBlock(ast)) {
      */
     public static Scope getSurroundingScope(DetailAST node) {
         Scope returnValue = null;
-        for (DetailAST token = node.getParent();
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java:134]: The only change in this PR right now should be here, we should hard code exactly what pitest is mutating this line to: `DetailAST token = node;`

**Comment by nrmancuso** [comment]
@Rohanraj123 IMO, we don’t need a custom projects file right now.

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java (hunk 1 above)
Comment (Rohanraj123) [reply src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java:134]: done !

**Comment by Rohanraj123** [comment]
> @Rohanraj123 IMO, we don’t need a custom projects file right now.

No worries I hardcoded the mutation and revert my changes that I made earlier.

**Pull request closed without merging**